src/stream/src/common/table/state_table.rs

```diff
@@ -808,22 +808,27 @@ where
         } else {
             Some(self.pk_serde.prefix(1))
         };
-        let range_end_suffix = watermark.map(|watermark| {
+        let watermark_suffix = watermark.map(|watermark| {
             serialize_pk(
                 row::once(Some(watermark)),
                 prefix_serializer.as_ref().unwrap(),
             )
         });
-        if let Some(range_end_suffix) = range_end_suffix {
-            let range_begin_suffix = vec![];
-            trace!(table_id = %self.table_id, range_end = ?range_end_suffix, vnodes = ?{
+        if let Some(watermark_suffix) = watermark_suffix {
+            let range_begin_suffix =
+                if prefix_serializer.unwrap().get_order_types()[0].nulls_are_largest() {
+                    vec![]
+                } else {
+                    vec![1u8]
```

Comment:
a little hack for the magic number :hot_face: can we call the memcomparable::encoding of the null?

Reply:
> a little hack for the magic number 🥵 can we call the memcomparable::encoding of the null?

fixed